### ISSUE TITLE
Videos causing Kodi to crash 

### ISSUE THREAD

**Issue by CHJ85** [Dis90/plugin.video.discoveryplus#84]
Possibly related to https://github.com/Dis90/plugin.video.discoveryplus/issues/83
A couple minutes in, the video stops and Kodi crashes.
```
2022-04-21 10:12:44.979 T:135      INFO <general>: Mixer feature: VDP_VIDEO_MIXER_FEATURE_NOISE_REDUCTION
2022-04-21 10:12:44.979 T:135      INFO <general>: Mixer feature: VDP_VIDEO_MIXER_FEATURE_SHARPNESS
2022-04-21 10:12:44.979 T:135      INFO <general>: Mixer feature: VDP_VIDEO_MIXER_FEATURE_DEINTERLACE_TEMPORAL
2022-04-21 10:12:44.979 T:135      INFO <general>: Mixer feature: VDP_VIDEO_MIXER_FEATURE_DEINTERLACE_TEMPORAL_SPATIAL
2022-04-21 10:12:44.980 T:135      INFO <general>: Mixer feature: VDP_VIDEO_MIXER_FEATURE_INVERSE_TELECINE
2022-04-21 10:12:44.980 T:135      INFO <general>: Mixer feature: VDP_VIDEO_MIXER_FEATURE_HIGH_QUALITY_SCALING_L1
2022-04-21 10:12:44.986 T:135      INFO <general>:  (VDPAU) screenWidth:1920 vidWidth:1920 surfaceWidth:1920
2022-04-21 10:12:44.986 T:135      INFO <general>:  (VDPAU) screenHeight:1080 vidHeight:1080 surfaceHeight:1088
2022-04-21 10:12:44.991 T:13       INFO <general>: CActiveAESink::OpenSink - initialize sink
2022-04-21 10:12:45.016 T:140      INFO <general>: COutput::OnStartup: Output Thread created
2022-04-21 10:12:45.016 T:141      INFO <general>: CMixer::OnStartup: Output Thread created
2022-04-21 10:12:45.016 T:141      INFO <general>:  (VDPAU) Creating the video mixer
2022-04-21 10:12:45.017 T:140      INFO <general>: VDPAU::COutput::InitBufferPool - Output Surface created
2022-04-21 10:12:45.049 T:7        INFO <general>: Skipped 6 duplicate messages..
2022-04-21 10:12:45.049 T:7        INFO <general>: CInteropState::Init: vdpau gl interop initialized
2022-04-21 10:12:45.049 T:7        INFO <general>: GL: Using VDPAU render method
2022-04-21 10:12:45.049 T:7        INFO <general>: GL: Using GL_ARB_pixel_buffer_object
2022-04-21 10:12:45.049 T:7        INFO <general>: Using GL_TEXTURE_2D
2022-04-21 10:12:45.193 T:13       INFO <general>: PulseAudio: Opened device Default in pcm mode with Buffersize 150 ms Periodsize 50 ms
2022-04-21 10:12:45.318 T:7        INFO <general>: Loading skin file: VideoFullScreen.xml, load type: KEEP_IN_MEMORY
2022-04-21 10:12:45.506 T:135      INFO <general>: CDVDVideoCodecFFmpeg::CDropControl: calculated diff time: 40000
2022-04-21 10:12:47.685 T:132      INFO <general>: AddOnLog: inputstream.adaptive: PosTime (62095)
2022-04-21 10:12:47.827 T:132      INFO <general>: AddOnLog: inputstream.adaptive: seekTime(62.1) for Stream:1 continues at 64.0 (PTS: 64000000)
2022-04-21 10:12:47.944 T:132      INFO <general>: AddOnLog: inputstream.adaptive: seekTime(64.0) for Stream:2 continues at 64.0 (PTS: 64000000)
2022-04-21 10:12:48.092 T:132      INFO <general>: AddOnLog: inputstream.adaptive: seekTime(64.0) for Stream:3 continues at 61.0 (PTS: 61000000)
2022-04-21 10:12:48.127 T:135      INFO <general>: CVDPAU::Release pre-cleanup
2022-04-21 10:12:48.128 T:141      INFO <general>: CMixer::OnExit: Output Thread terminated
2022-04-21 10:12:48.171 T:135      INFO <general>:  (VDPAU) screenWidth:1920 vidWidth:1920 surfaceWidth:1920
2022-04-21 10:12:48.171 T:135      INFO <general>:  (VDPAU) screenHeight:1080 vidHeight:1080 surfaceHeight:1088
2022-04-21 10:12:48.202 T:143      INFO <general>: COutput::OnStartup: Output Thread created
2022-04-21 10:12:48.202 T:144      INFO <general>: CMixer::OnStartup: Output Thread created
2022-04-21 10:12:48.202 T:144      INFO <general>:  (VDPAU) Creating the video mixer
2022-04-21 10:12:48.203 T:143      INFO <general>: VDPAU::COutput::InitBufferPool - Output Surface created
2022-04-21 10:12:48.235 T:7        INFO <general>: Skipped 6 duplicate messages..
2022-04-21 10:12:48.235 T:7        INFO <general>:  (VDPAU) Close
2022-04-21 10:12:48.235 T:7        INFO <general>:  (VDPAU) FiniVDPAUOutput
2022-04-21 10:12:48.252 T:140      INFO <general>: COutput::OnExit: Output Thread terminated
2022-04-21 10:12:48.256 T:7        INFO <general>: CInteropState::Init: vdpau gl interop initialized
2022-04-21 10:12:48.256 T:7        INFO <general>: GL: Using VDPAU render method
2022-04-21 10:12:48.256 T:7        INFO <general>: GL: Using GL_ARB_pixel_buffer_object
2022-04-21 10:12:48.256 T:7        INFO <general>: Using GL_TEXTURE_2D
2022-04-21 10:12:51.514 T:132      INFO <general>: AddOnLog: inputstream.adaptive: PosTime (77017)
2022-04-21 10:12:51.686 T:132      INFO <general>: AddOnLog: inputstream.adaptive: seekTime(77.0) for Stream:1 continues at 78.0 (PTS: 78000000)
2022-04-21 10:12:51.818 T:132      INFO <general>: AddOnLog: inputstream.adaptive: seekTime(78.0) for Stream:2 continues at 78.0 (PTS: 77994666)
2022-04-21 10:12:51.818 T:132      INFO <general>: AddOnLog: inputstream.adaptive: seekTime(78.0) for Stream:3 continues at 75.4 (PTS: 75360000)
2022-04-21 10:12:52.056 T:135      INFO <general>: CVDPAU::Release pre-cleanup
2022-04-21 10:12:52.058 T:144      INFO <general>: CMixer::OnExit: Output Thread terminated
2022-04-21 10:12:52.121 T:135      INFO <general>:  (VDPAU) screenWidth:1920 vidWidth:1920 surfaceWidth:1920
2022-04-21 10:12:52.121 T:135      INFO <general>:  (VDPAU) screenHeight:1080 vidHeight:1080 surfaceHeight:1088
2022-04-21 10:12:52.149 T:145      INFO <general>: COutput::OnStartup: Output Thread created
2022-04-21 10:12:52.149 T:146      INFO <general>: CMixer::OnStartup: Output Thread created
2022-04-21 10:12:52.149 T:146      INFO <general>:  (VDPAU) Creating the video mixer
2022-04-21 10:12:52.152 T:145      INFO <general>: VDPAU::COutput::InitBufferPool - Output Surface created
2022-04-21 10:12:52.198 T:7        INFO <general>: Skipped 6 duplicate messages..
2022-04-21 10:12:52.198 T:7        INFO <general>:  (VDPAU) Close
2022-04-21 10:12:52.198 T:7        INFO <general>:  (VDPAU) FiniVDPAUOutput
2022-04-21 10:12:52.214 T:143      INFO <general>: COutput::OnExit: Output Thread terminated
2022-04-21 10:12:52.219 T:7        INFO <general>: CInteropState::Init: vdpau gl interop initialized
2022-04-21 10:12:52.219 T:7        INFO <general>: GL: Using VDPAU render method
2022-04-21 10:12:52.219 T:7        INFO <general>: GL: Using GL_ARB_pixel_buffer_object
2022-04-21 10:12:52.219 T:7        INFO <general>: Using GL_TEXTURE_2D
2022-04-21 10:12:54.467 T:132      INFO <general>: AddOnLog: inputstream.adaptive: PosTime (89860)
2022-04-21 10:12:54.649 T:132      INFO <general>: AddOnLog: inputstream.adaptive: seekTime(89.9) for Stream:1 continues at 90.0 (PTS: 90000000)
2022-04-21 10:12:54.650 T:132      INFO <general>: AddOnLog: inputstream.adaptive: seekTime(90.0) for Stream:2 continues at 90.0 (PTS: 89984000)
2022-04-21 10:12:54.796 T:132      INFO <general>: AddOnLog: inputstream.adaptive: seekTime(90.0) for Stream:3 continues at 90.0 (PTS: 90000000)
2022-04-21 10:12:54.965 T:135      INFO <general>: CVDPAU::Release pre-cleanup
2022-04-21 10:12:54.968 T:146      INFO <general>: CMixer::OnExit: Output Thread terminated
2022-04-21 10:12:55.026 T:135      INFO <general>:  (VDPAU) screenWidth:1920 vidWidth:1920 surfaceWidth:1920
2022-04-21 10:12:55.026 T:135      INFO <general>:  (VDPAU) screenHeight:1080 vidHeight:1080 surfaceHeight:1088
2022-04-21 10:12:55.054 T:147      INFO <general>: COutput::OnStartup: Output Thread created
2022-04-21 10:12:55.054 T:148      INFO <general>: CMixer::OnStartup: Output Thread created
2022-04-21 10:12:55.054 T:148      INFO <general>:  (VDPAU) Creating the video mixer
2022-04-21 10:12:55.057 T:147      INFO <general>: VDPAU::COutput::InitBufferPool - Output Surface created
2022-04-21 10:12:55.104 T:7        INFO <general>: Skipped 6 duplicate messages..
2022-04-21 10:12:55.104 T:7        INFO <general>:  (VDPAU) Close
2022-04-21 10:12:55.104 T:7        INFO <general>:  (VDPAU) FiniVDPAUOutput
2022-04-21 10:12:55.120 T:145      INFO <general>: COutput::OnExit: Output Thread terminated
2022-04-21 10:12:55.125 T:7        INFO <general>: CInteropState::Init: vdpau gl interop initialized
2022-04-21 10:12:55.125 T:7        INFO <general>: GL: Using VDPAU render method
2022-04-21 10:12:55.126 T:7        INFO <general>: GL: Using GL_ARB_pixel_buffer_object
2022-04-21 10:12:55.126 T:7        INFO <general>: Using GL_TEXTURE_2D

```

**Comment by Dis90** [comment]
Probably same issue. What OS and Kodi version you are using?

**Comment by CHJ85** [comment]
Oh sorry, I should have mentioned.
Kodi 19.2 Matrix, Manjaro Linux.

**Comment by Dis90** [comment]
Try to update latest Kodi (19.4) and InputStream Adaptive 19.0.3

**Comment by CHJ85** [comment]
Okay. Not sure if the version is related to this or not, as I haven't had issues with other addons.
I installed Kodi through flatpak, so I'm gonna have to wait for that build to catch up.

**Comment by Dis90** [comment]
It is always good to check with latest version but probably doesn't fix the issue. Some users on Kodi forums have reported that videos play without crash on Kodi 20.

**Comment by Dis90** [comment]
I downgraded to Kodi 19 and I was able to get Kodi to crash. Currently I don't have any ideas why this is happening. Tried to disable DRM and InputStream Adaptive but still Kodi crashes.

**Comment by CHJ85** [comment]
Okay thank you so much for your help.
But just to avoid confusion and for future references. What version does work? Because the latest version I'm able to find is 19.4 from the official Arch repo and 19.2 from Flathub which is the one I'm currently on.

4

**Comment by Dis90** [comment]
Sorry it is not possible disable subtitles on add-on. All video playback related is handled by InputStream Adaptive. What InputStream Adaptive version do you have? Some subtitle issues was fixed in version 19.0.2

**Comment by CHJ85** [comment]
That's the version I got installed. 19.0.2.
I find it extremely odd that I cannot exclude subtitles from InputStream. It allows me to stream only video and only audio and video with subtitles, but not video+audio. Weird. I should probably open up an issue regarding this on their github.

**Comment by Dis90** [comment]
Yep please open issue to InputStream Adaptive Github :)
This video+audio problem and crashing is something I can't fix. And it would be nice if someone who are experiencing the problem opens issue to ISA Github.

**Comment by Dis90** [comment]
This is now fixed in InputStream Adaptive 19.0.4. You have to wait until your OS devs release new version. In meantime if you don't need subtitles you can disable those from osd.

**Comment by CHJ85** [comment]
Thank you so much for all your great work.
